api/src/main/scala/org/ensime/api/common.scala

```diff
@@ -23,9 +23,10 @@ object DeclaredAs {
   case object Object extends DeclaredAs('object)
   case object Class extends DeclaredAs('class)
   case object Field extends DeclaredAs('field)
+  case object TypeAlias extends DeclaredAs('typeAlias)
```

Comment:
this needs a bump in the protocol version. I'm not sure we want to expose type aliases at the api level.